Singleton.cpp

```diff
@@ -0,0 +1,36 @@
+#include <iostream>
```

Comment:
Please see `Meyers Singleton`

The beauty of the Meyers Singleton in C++11 is that it's automatically thread-safe. That is guaranteed by the standard: [Static variables with block scope.](https://www.modernescpp.com/index.php/thread-safe-initialization-of-data)

Reply:
```
class MySingleton
{
public:
  static MySingleton& getInstance(){
    static MySingleton instance;
    // volatile int dummy{};
    return instance;
  }
private:
  MySingleton()= default;
  ~MySingleton()= default;
  MySingleton(const MySingleton&)= delete;
  MySingleton& operator=(const MySingleton&)= delete;
};
```